### PR TITLE
Add autocomplete

### PR DESCRIPTION
**Pull request by Gamgeee**
Add members edit

### CODE REVIEW

File: src/components/Autocomplete/index.tsx [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    const timer = setTimeout(() => {
+      setIsLoading(true);
+
+      suggestionsSource(v).then(data => {
+        setSuggestions(data);
+        setIsLoading(false);
+      }, () => {
+        setIsLoading(false);
+      });
+    }, SEARCH_DELAY_MS);
```
Comment (doskachok) [inline src/components/Autocomplete/index.tsx]: maybe store timer in useRef instead of state

File: src/components/Autocomplete/index.tsx [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+  const handleSuggestionSelected = (suggestion: any) => {
+    onSuggestionSelected(suggestion);
+  };
+
+  return (
+    <Wrapper fullWidth>
+      <AutocompleteInput
+        placeholder={placeholder}
+        value={value}
+        name='autocomplete'
```
Comment (doskachok) [inline src/components/Autocomplete/index.tsx]: move name to props with default value 'autocomplete'

File: src/components/Autocomplete/index.tsx [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+import { useTranslation } from 'react-i18next';
+
+import { AutocompleteInput, LoadingRow, NoResultsRow, SearchIcon, SuggestionRow, SuggestionsWrapper, Wrapper } from './index.styled';
+import { IAutocompleteSuggestion } from './types';
+import SearchIconSVG from '../../assets/images/search-icon.svg';
+
+interface Props {
+  placeholder?: string;
+  onSuggestionSelected: (suggestion: any) => void;
+  suggestionTemplate?: (suggestion: any) => ReactElement<any>;
```
Comment (doskachok) [inline src/components/Autocomplete/index.tsx]: use IAutocompleteSuggestion

File: src/features/groups/Members/AddMemberModal/index.tsx [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+  const suggestionTemplate = useCallback((suggestion: IUser) => {
+    return <UserRow reversedTheme={true} user={suggestion} />
+  }, []);
+
+  const usersSearchSource = (searchQuery: string) => new Promise<IUser[]>((resolve, reject) => {
+    getUsers(searchQuery)
+      .then(data => resolve(data.data || []))
+      .catch(err => reject(err));
+  });
```
Comment (doskachok) [inline src/features/groups/Members/AddMemberModal/index.tsx:36]: Can we just use getUsers without wrapper as suggestionsSource?

File: src/features/groups/Members/AddMemberModal/index.tsx [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+  useEffect(() => {
+    getKnownsUsers();
+  }, [getKnownsUsers]);
```
Comment (doskachok) [inline src/features/groups/Members/AddMemberModal/index.tsx]: redundant useEffect, use not lazy query

File: src/features/groups/Members/index.tsx [hunk 6]
```diff
@@ -1,61 +1,100 @@
...
 
-  }, []);
+  useEffect(() => {
+    return () => {
+      if (addMemberModalId) {
+        modalContext.closeModal(addMemberModalId);
+        setAddMemberModalId(null);
+      }
+    };
+  }, [addMemberModalId, modalContext]);
```
Comment (doskachok) [inline src/features/groups/Members/index.tsx:60]: Try to test with addMemberModalId as ref

**Review by doskachok — CHANGES_REQUESTED** [review]

File: src/features/groups/Members/AddMemberModal/index.tsx (hunk 4 above)
Comment (Gamgeee) [reply src/features/groups/Members/AddMemberModal/index.tsx:36]: No, type incompatible error occur

File: src/features/groups/Members/index.tsx (hunk 6 above)
Comment (Gamgeee) [reply src/features/groups/Members/index.tsx:60]: Ref cause modal to be closed after it opened

**Comment by doskachok** [comment]
Reimplemented in #57

**Pull request closed without merging**